### DISCUSSION TITLE
Maximum LED count

### DISCUSSION THREAD

**Discussion by sventertainment** [Makuna/NeoPixelBus#489] [Q&A] [answered]
Hi Makuna,

First of all, thank you very much for this great library! It helped me a lot, especially using the NeoTopology for mapping the LEDs.

My question...
What is the maximum number of LEDs that is supported by the NeoPixelBus library? I'm using SK6812 RGBW LEDs on an Arduino MKR ZERO. Now using 600 LEDs (powered by applying 5V every 100 LEDs) which is working great (without complex animations), but requesting a multiple of 600.

Is it possible to run multiple instances of the library (on multiple pins) to have more LEDs?

Hope you can help me with this.
Thank you!

Best regards,
Sven.

**Comment by Makuna** [comment, marked as answer]
[See this Wiki FAQ on the topic](https://github.com/Makuna/NeoPixelBus/wiki/FAQ-%237)  

Max count is limited by memory in most cases.  RGBW LEDs will require 4 bytes per pixel, so 600 x 4 = 2400bytes to store the strip data.  Looks like the MKRZero has 32K.  If you use the Arduino IDE, when you compile your sketch it will give your the static memory used by your sketch and what is left over, the memory for the LEDs will be allocated using the left over.  But other libraries may also use dynamic allocation; check with them.

```
Global variables use 2104 bytes (6%) of dynamic memory, leaving 30664 bytes for local variables. Maximum is 32768 bytes.
```

You can do multiple strips, but since it is bitbang method (meaning no special hardware support and that its synchronous) there is little benefit to use it unless it helps you with logic of your code or wiring of the strips.

Also note, the larger the strip, the longer it takes to send the data to it.  Most LEDs it will take a fixed amount of time per byte, [see this wiki FAQ on the topic](https://github.com/Makuna/NeoPixelBus/wiki/FAQ-%235)
